gulp/plugins/gulp-component-info.js

```diff
@@ -0,0 +1,36 @@
+import gutil from 'gulp-util'
+import through from 'through2'
+import Vinyl from 'vinyl'
+
+import { generateComponentIndex } from './util'
+
+const pluginName = 'gulp-component-info'
+
+export default () => {
+  const files = {}
+
+  return through.obj(
+    (file, enc, cb) => {
+      if (file.isNull()) {
+        cb(null, file)
+        return
+      }
+
+      if (file.isStream()) {
+        cb(new gutil.PluginError(pluginName, 'Streaming is not supported'))
+        return
+      }
+
+      files[file.basename] = JSON.parse(file.contents)
```

Comment:
I think we should generate separate files for each component.  This will help us with code splitting.  It also helps for navigating around the generated code.

See how Stardust creates JSON files for various concerns, per component, in the `docs/src` dir after running `yarn build:docs`.

Reply:
A little confusing, but this plugin just generates `index.js` in `docs/src/componentInfo` which will really help us with code splitting :) It replaces `require.context()` with valid imports.

```js
import Confirm from "./Confirm.info.json";
import MountNode from "./MountNode.info.json";
import Pagination from "./Pagination.info.json";
import PaginationItem from "./PaginationItem.info.json";

// etc...
```